src/ssl.c

```diff
@@ -16286,8 +16286,9 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
             }
 
             XFREE(bio, 0, DYNAMIC_TYPE_OPENSSL);
+            return WOLFSSL_SUCCESS;
         }
-        return 1;
+        return WOLFSSL_FAILURE;
```

Comment:
Are you sure this is correct?  Why is bio == NULL a failure?

Reply:
openSSL1.1.1 returns 0 as following:
 
```
int BIO_free(BIO *a)
{
    int ret;

    if (a == NULL)
        return 0;

```